solution/backend/common/serializers/mix.py

```diff
@@ -15,5 +16,8 @@ def get_locations(self, obj):
     @extend_schema_field(MetaCategorySerializer.many(False))
     def get_category(self, obj):
         if self.context['request'].GET.get("category_details") == 'true':
-            return AbstractCategoryPolymorphicSerializer(obj.category).data
+            if obj.category:
```

Comment:
This will allow category to be interchangeable on the serializer without having to do any changes on the front end to determine where to get the category from.